src/pop_protocol.erl

```diff
@@ -273,3 +326,37 @@ get_verfier_next_block_time(PD, VerifierIndex)
 	    NextTime
     end.
     
+%% @doc Creates next block in the chain for a given verifier.
+%% 
+%% Created block is unsigned.
+
+generate_new_block(VerifierIndex, PD)
```

Comment:
So are you suppose to generate a new block first and then add transaction to it afterwards?

Can you clarify in comments?

Reply:
Blocktree is responsible for putting pending transactions into the block, you'd never do it directly